docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,226 @@
+- Status:
+- Date: 2023-06-07
+- Author: [@brizental](https://github.com/brizental)
+- RFC PR: [#7112](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7112)
+- Implementation GitHub issue: TODO
+
+## Problem Statement
+
+The VPN application is divided in two processes: the main application and the daemon. The main application is where the user interface runs and the daemon is what manages the VPN configuration and tunnel.
+
+In desktop environments, even while backgrounded the main application process is still running, but in mobile environments it is not. Due to this caveat, the mobile daemon processes have a separate instance of Glean to be able to collect telemetry about the connection while the application is in the background.
+
+This architecture poses a problem for analyzers. It is not possible to easily join the data from the same session between daemon and main application. The session ids are different due to these being two different Glean instances.
+
+For more information on this topic, refer to [Mozilla VPN Telemetry Refactor](https://docs.google.com/document/d/1jyNZ_g_cUpZZsEr2hYwnwZgkxlvqTmoFUJKp_LAOPts/edit#heading=h.xararkvsnpss).
+
+## Proposed Solution
+
+> **Note** Honestly, I think for this proposed solution it's just easier to look at the pseudo-code in Appendix 1.
+
+A new metric will be added to both the daemonsession and the session pings: shared_session_id. This metric will have lifetime ping i.e. it is cleared from storage whenever a ping containing it is submitted. Thus it needs to be set again every time a new session/daemonsession ping is submitted.
+
+Both daemon and main application will store the value of this metric in their local storages for synchronization purposes. More about this in the sections below.
+
+A new reason will be added to the session pings “flush_init”. As better described in the sections below, the processes will be able to identify that either the VPN tunnel or main application did not gracefully hit the deactivation signal i.e. were force killed. The “flush_init” ping will be used to flush any data remaining from the previous session.
+
+There will be four key triggers that will handle the shared_session_id: initialization, on VPN activation, ping timer and on VPN deactivation.
```

Comment:
Is it possible that Daemon or the client could crash independently of each other and lose their session id? If that happens, I imagine a new session id is creating once it's restarted, how do we ensure that the other party knows about this and maintains the relevance somehow?

Reply:
Hm, need to think if this still applied on the rewritten proposal.

Reply:
> Is it possible that Daemon or the client could crash independently of each other and lose their session id? If that happens, I imagine a new session id is created once it's restarted, how do we ensure that the other party knows about this and maintains the relevance somehow?

Ok so, rereading this. It is not possible. Not in the rewritten or in the original proposal. If either party crashes, the session is not lost, because it is in permanent storage.